transport/src/main/java/io/netty/channel/socket/nio/NioSocketChannel.java

```diff
@@ -414,13 +414,13 @@ protected void doWrite(ChannelOutboundBuffer in) throws Exception {
                     // Zero length buffers are not added to nioBuffers by ChannelOutboundBuffer, so there is no need
                     // to check if the total size of all the buffers is non-zero.
                     // We limit the max amount to int above so cast is safe
-                    long attemptedBytes = in.nioBufferSize();
                     final long localWrittenBytes = ch.write(nioBuffers, 0, nioBufferCnt);
                     if (localWrittenBytes <= 0) {
                         incompleteWrite(true);
                         return;
                     }
                     // Casting to int is safe because we limit the total amount of data in the nioBuffers to int above.
+                    long attemptedBytes = in.nioBufferSize();
```

Comment:
sorry this is also incorrect... nioBufferSize() will also not produce the correct value after write was called.

Reply:
is this attemptedBytes incorrect in default case ?
this is the value in ChannelOutboundBuffer and only be changed in `in.nioBuffers(1024, maxBytesPerGatheringWrite);`.
And the ch.write just write this ByteBuffer[] into channel, is this will change the attemptedBytes value?

Reply:
I just debug here and get the same value between before write and after write, in whick case the attemptedBytes will changed? I will have a try :>